rosapi/src/rosapi/glob_helper.py

```diff
@@ -30,7 +32,10 @@ def get_param(param_name):
 def filter_globs(globs, full_list):
     # If the globs are empty (weren't defined in the params), return the full list
     if globs is not None and len(globs) > 0:
-        return filter(lambda x: any_match(x, globs), full_list)
+        if PYTHON2:
+            return filter(lambda x: any_match(x, globs), full_list)
+        else:
+            return [i for i in filter(lambda x: any_match(x, globs), full_list)]
```

Comment:
Instead of having all this logic specific to Python version, the Python 3 path should work fine in Python 2 and should be the only path if that is the case.